packages/react-avatar/SPEC.md

```diff
@@ -0,0 +1,272 @@
+    borderRadius: '0',
+  }}
+/>
+```
+
+Where hexagonSvg is defined as:
+
+```js
+hexagonSvg =
+  'data:image/svg+xml;utf8,' +
+  '<svg width="36" height="32" viewBox="0 0 36 32" fill="none" xmlns="http://www.w3.org/2000/svg">' +
+  '<path fill="rgb(232,232,232)" d="M0.407926 17.528C-0.135976 16.5859 -0.135975 15.4141 0.407926 14.472' +
+  'L7.91541 1.46793C8.44076 0.557947 9.39444 0 10.4245 0H25.5755C26.6056 0 27.5592 0.557951 28.0846 1.46793' +
+  'L35.5921 14.472C36.136 15.4141 36.136 16.5859 35.5921 17.528L28.0846 30.5321' +
+  'C27.5592 31.4421 26.6056 32 25.5755 32H10.4245C9.39443 32 8.44076 31.4421 7.91541 30.5321L0.407926 17.528Z"/>' +
+  '</svg>';
+```
+
+## API
+
+From [Avatar.types.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-avatar/src/components/Avatar/Avatar.types.tsx):
+
+### Slots
+
+- `image` - The Avatar's image, if available.
+- `label` - The text shown when there's no image. Defaults to the initials derived from `name` using `getInitials`.
+- `icon` - Icon displayed when there's no image or intials available.
+- `badge` - Badge to show the avatar's status.
+
+### Props
+
+```ts
+export interface AvatarProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /** The Avatar's image. */
+  image?: ShorthandValue<{}>;
+
+  /** The label shown when there's no image. Defaults to the initials derived from `name` using `getInitials`. */
+  label?: ShorthandValue<{}>;
+
+  /** Icon displayed when there's no image or intials available, or if `display="icon"`. */
+  icon?: ShorthandValue<{}>;
+
+  /** What the avatar displays. This can be used to override the default behavior, for example to show the icon even
+   *  if the avatar has initials that could be shown as the label. */
+  display?: 'image' | 'label' | 'icon';
+
+  /**
+   * Optional activity indicator
+   * * true: the avatar will be decorated according to activeDisplay
+   * * false: the avatar will be reduced in size and partially transparent
+   * * undefined (default): normal display
+   */
+  active?: boolean;
+
+  /**
+   * The type of visual treatment to use when `active="true"`
+   * @defaultvalue ring
+   */
+  activeDisplay?: 'ring' | 'shadow' | 'glow' | 'ring-shadow' | 'ring-glow';
+
+  /** Badge to show the avatar's status. */
+  badge?: ShorthandValue<BadgeProps>;
+
+  /** The name used for displaying the initials of the avatar if the image is not provided. */
+  name?: string;
+
+  /** The avatar can have a square shape. */
+  square?: boolean;
+
+  /** Size of the avatar */
+  size?: AvatarSizeValue;
+
+  /**
+   * If a non-standard size is needed, use customSize instead of size.
+   *
+   * The dimensions of the avatar will be the given customSize, and the icon
+   * and font sizes will be based on the next-smaller AvatarSizeValue.
+   * For more fine-grained control over the font and icon sizes, use tokens:
+   * `tokens={{ fontSize: '...px', fontWeight: '...', iconSize: '...px' }}`
+   */
+  customSize?: number;
```

Comment:
I wanted to get feedback on the use of `size` as a restricted set of available sizes, specified by design, and `customSize` as a "relief valve" for anyone who really needs a different size. The restricted set of sizes was requested by design.

The two props both end up determining the pixel size of the Avatar in the end, but the `customSize` prop makes it more explicit that you're not picking an "approved" size. It also makes it possible to do something different in the future if we decide to deprecate custom sizes.

Anyone have opinions about this?

Reply:
Should we this via `tokens` instead?

Reply:
I originally had it use `tokens` (which map to css variables), but the design spec has a bunch of other properties that change based on the size of the avatar. For example, the icon size is "small" if the avatar size is <= 24, and "medium" if the size is >24 and <=40, etc. As far as I can tell, that kind of conditional logic based on the value of a css variable isn't possible without using javascript.

Therefore, it made more sense to have the size be specified in a javascript variable.

Reply:
Got it. I am voting to remove this prop as it potentially creates inconstancies:
- how this pattern can be used for other components?
- we have tokens and IMO they should be a single source of truth
- customers should mostly use predefined values as they defined by the design team

Reply:
I agree with @layershifter here.

I think the following should happen:
- The `size` prop can stay as a ramp that controls many things.
- We should remove the `customSize` prop and have indications on how, if a user wants to customize the component, they can still do it by specifying all tokens individually.

Reply:
Ok that seems reasonable. So if you wanted it to be e.g. 42px, you'd write:
`<Avatar size={40} tokens={{ width: '42px', height: '42px' }} />`

I'll update the spec and log an issue to update the implementation.